src/gui/qgsmapcanvas.h

```diff
@@ -1422,6 +1423,8 @@ class GUI_EXPORT QgsMapCanvas : public QGraphicsView, public QgsExpressionContex
 
     std::unique_ptr< QgsTemporaryCursorOverride > mTemporaryCursorOverride;
 
+    QMap <QString, QDateTime> mRendererErrors;
```

Comment:
It would be nice to document that we are trying not to throw rendering errors on every render, that's why we keep the list of recent errors with time